### PR TITLE
mqtt: add test case for protocol string keyword

### PR DESCRIPTION
**Pull request by satta**
This PR adds a new MQTT test that checks the new `mqtt.connect.protocolstring` keyword that is currently not tested anywhere else.

We only run the test when the associated detection source is available in the tested source tree.

## Ticket

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/6396

### CODE REVIEW

**Comment by catenacyber** [comment]
Requires https://github.com/OISF/suricata/pull/9600 to run and not be skipped

File: tests/mqtt-connect-rules/test.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+requires:
+  features:
+    - HAVE_LIBJANSSON
+  files:
+    - rust/src/mqtt/parser.rs
```
Comment (catenacyber) [inline tests/mqtt-connect-rules/test.yaml:5]: We can just do with the first one

File: tests/mqtt-connect-rules/test.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+requires:
+  features:
+    - HAVE_LIBJANSSON
```
Comment (catenacyber) [inline tests/mqtt-connect-rules/test.yaml:3]: We do not need to specify this feature anymore

File: tests/mqtt-connect-rules/test.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      count: 1
+      match:
+        event_type: alert
+        alert.signature: MQTT CONNECT protocol string SUCCESS
+
+  - filter:
+      count: 1
+      match:
+        event_type: alert
+        alert.signature: MQTT CONNECT protocol string SUCCESS2
```
Comment (catenacyber) [inline tests/mqtt-connect-rules/test.yaml:60]: Should you test the absence of  sid:3 ?

File: tests/mqtt-connect-rules/test.yaml (hunk 1 above)
Comment (satta) [reply tests/mqtt-connect-rules/test.yaml:5]: Hmm, the first one? I'd say that the presence of `src/detect-mqtt-connect-protocol-string.c` should be enough to indicate that the feature can be tested. I'll leave that one in.

Comment (catenacyber) [reply tests/mqtt-connect-rules/test.yaml:5]: Indeed. I meant just keep `src/detect-mqtt-connect-protocol-string.c`

File: tests/mqtt-connect-rules/test.yaml (hunk 3 above)
Comment (satta) [reply tests/mqtt-connect-rules/test.yaml:60]: Done

**Comment by satta** [comment]
Next PR: #1459

**Pull request closed without merging**